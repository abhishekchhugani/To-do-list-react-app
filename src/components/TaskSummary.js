//Component to display count of total to do items and completed to do items

import React from 'react';

const TaskSummary = ({ todos }) => {

    const todo = todos.length ? (
        todos.filter(todo => {
            return todo.complete;
        })
    ) : []
    
    return (
        <div className="task-summary mb20">
            <div className="row inner-height">
                <div className="col s12 valign-wrapper horizontal-center total" id="totalCount">
                    Total: {todos.length}
                </div>
                <div className="col s12 valign-wrapper horizontal-center" id="completedCount">
                    Completed : {todo.length}
                </div>
            </div>
        </div>
    )
}

export default TaskSummary;


