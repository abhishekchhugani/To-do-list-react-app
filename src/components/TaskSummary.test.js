import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import TaskSummary from './TaskSummary';

  const todos = [
    {
        id: 1,
        content: 'Hello',
        complete: true
    },
    {
        id: 2,
        content: 'World',
        complete: false
    }
]

it('renders without crashing', () => {
    shallow(<TaskSummary  todos={todos} />);
});

it('snapshot test', () => {
    const wrapper = shallow(<TaskSummary todos={todos}/>);
    expect(wrapper).toMatchSnapshot();
});


it('Displays total count', () => {
    const wrapper = shallow(<TaskSummary todos={todos}/>);
    expect(wrapper.find('#totalCount'))
});

it('Displays completed count', () => {
    const wrapper = mount(<TaskSummary todos={todos}/>);
    expect(wrapper.find('#completedCount'))
});