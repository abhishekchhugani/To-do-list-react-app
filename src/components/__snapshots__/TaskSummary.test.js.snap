// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskSummary
    todos={
        Array [
            Object {
              "complete": true,
              "content": "Hello",
              "id": 1,
            },
            Object {
              "complete": false,
              "content": "World",
              "id": 2,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row inner-height"
>
        <div
                className="col s12 valign-wrapper horizontal-center total"
                id="totalCount"
        >
                Total: 
                2
        </div>
        <div
                className="col s12 valign-wrapper horizontal-center"
                id="completedCount"
        >
                Completed : 
                1
        </div>
</div>,
      "className": "task-summary mb20",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col s12 valign-wrapper horizontal-center total"
            id="totalCount"
>
            Total: 
            2
</div>,
          <div
            className="col s12 valign-wrapper horizontal-center"
            id="completedCount"
>
            Completed : 
            1
</div>,
        ],
        "className": "row inner-height",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total: ",
              2,
            ],
            "className": "col s12 valign-wrapper horizontal-center total",
            "id": "totalCount",
          },
          "ref": null,
          "rendered": Array [
            "Total: ",
            2,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Completed : ",
              1,
            ],
            "className": "col s12 valign-wrapper horizontal-center",
            "id": "completedCount",
          },
          "ref": null,
          "rendered": Array [
            "Completed : ",
            1,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row inner-height"
>
          <div
                    className="col s12 valign-wrapper horizontal-center total"
                    id="totalCount"
          >
                    Total: 
                    2
          </div>
          <div
                    className="col s12 valign-wrapper horizontal-center"
                    id="completedCount"
          >
                    Completed : 
                    1
          </div>
</div>,
        "className": "task-summary mb20",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col s12 valign-wrapper horizontal-center total"
              id="totalCount"
>
              Total: 
              2
</div>,
            <div
              className="col s12 valign-wrapper horizontal-center"
              id="completedCount"
>
              Completed : 
              1
</div>,
          ],
          "className": "row inner-height",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Total: ",
                2,
              ],
              "className": "col s12 valign-wrapper horizontal-center total",
              "id": "totalCount",
            },
            "ref": null,
            "rendered": Array [
              "Total: ",
              2,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Completed : ",
                1,
              ],
              "className": "col s12 valign-wrapper horizontal-center",
              "id": "completedCount",
            },
            "ref": null,
            "rendered": Array [
              "Completed : ",
              1,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
